test/new-e2e/tests/containers/ecs_test.go

```diff
@@ -94,3 +66,117 @@ func TestECSSuite(t *testing.T) {
 		endTime.UnixMilli(),
 	))
 }
+
+type ecsSuite struct {
+	baseSuite
+
+	ecsClusterName string
+}
+
+func (suite *ecsSuite) TestNginx() {
+	// `nginx` check is configured via docker labels
+	// Test it is properly scheduled
+	suite.testMetric("nginx.net.request_per_s",
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered having an helper function instead of using a `suite` as a base ? Something like 
```golang
func assertMetrics(t *testing.T, fakeintake *fakeintake.Client, metrics MetricsToCheck)
``` 

Not a critic, I am curious if you considered it as an alternative, as here we use the suite to pass the testing and fakeintake context rather than to leverage the suite interface.

An alternative could also be to pass a `testing.T` context to `fakeintake.Client`. But I can see this has the limit of not allowing switching testing context when calling it from an `Eventually` or from a `t.Run` that create a separate testing context.

Reply:
To be honest, I’m not convinced by the current way the helper functions are designed. And I know I’ll refactor them at some point.

For ex., the `testMetrics` function takes a list of expected tags in the form of regular expressions. I think the code might be less painful to write if the calls to `regexp.MustCompile(…)`  were done inside the `testMetrics(…)` function itself rather than on the caller side, everywhere this function is invoked.

In the current tests, `testMetrics(…)` is only checking for the tags on the metrics.
Its parameters are:
* the name of the metric to look at
* a list of tags to filter the series to consider
* the exhaustive list of tags we expect to have on the series.
At some point, we’ll want to potentially check the value as well. As values are float, it will be an expected range.
It means two more parameters:
* expected minimum value
* expected maximum value.

Having all that as positional arguments will become painful.
So, we might want to have named arguments, similar to the Pulumi API, with some arguments being optional:
```go
assertMetric(&assertMetricArgs{
  filter: {
    metricName: …,
    tags: …,
  },
  expect: {
    tags: …,
    value: {
      min: …,
      max: …,
})
```

About using a helper function instead of a `baseSuite` method, I don’t have any strong opinion. It permitted to pass less parameters as the fakeintake object was a field of `baseSuite`.
I think that I’d like to keep the fact that `assertMetric` creates a sub-test with `suite.Run(…)` because it permits to have a more granular report.

So, yeah, I agree that this needs improvement.

Reply:
I also like how `suite.Run` or `t.Run` improve readability of reports, I do not use them for the separate context, still it can be misleading to use the parent context instead of the current one.

Thinking if we can have this in a generic `FilterMetrics` inside the fakeintake, or if we can have some external aggregation helpers. 

What `ddev` does in integrations-core tests is having an aggregator that marks each checked metric and eventually can provide all metrics that were not checked with `agg.RequireAllMetricsChecked`.